### PR TITLE
Media gallery bugfixes

### PR DESCRIPTION
**Pull request by lhoffbeck**
### PR Summary: 

1. Fixes bug where media flashes after switching to variant with the same featured image
2. Fixes bug where featured media is removed if switching from a variant with featured media to a variant without featured media

### What approach did you take?

1. Add check that the featured media is not already the first child of the parent before prepending it
2. Abort the media update if the variant has no featured media

### Visual impact on existing themes

In both cases, reverts us to the behavior that was present in V14.0.0

### Testing steps/scenarios

Use theme [lh-media-gallery-bugfixes](https://os2-demo.myshopify.com/?_ab=0&_fd=0&_sc=1&key=426fb1f420e906418fa2448926c74e347c9832a999315511ee1be9880525803a&preview_theme_id=163385212950).

- [ ] [On this product](https://os2-demo.myshopify.com/products/louise-slide-sandal)
  - [ ] Check that switching size does not cause the main image to flash
  - [ ] Check that the slider is reset after switching variants.
  - [ ] Check that switching color updates the featured image and thumbnail
- [ ] On [this product](https://os2-demo.myshopify.com/products/combine-listing-test-orange)
  - [ ] Check that switching from small to medium does not remove the variant featured media

### Checklist
- [x] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [x] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [x] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [x] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: assets/media-gallery.js [hunk 1]
```diff
@@ -41,11 +41,13 @@ if (!customElements.get('media-gallery')) {
...
+          activeMedia.parentElement.firstChild !== activeMedia && activeMedia.parentElement.prepend(activeMedia);
+
           if (this.elements.thumbnails) {
             const activeThumbnail = this.elements.thumbnails.querySelector(`[data-target="${mediaId}"]`);
-            activeThumbnail.parentElement.prepend(activeThumbnail);
+            activeThumbnail.parentElement.firstChild !== activeThumbnail && activeThumbnail.parentElement.prepend(activeThumbnail);
           }
+
           if (this.elements.viewer.slider) this.elements.viewer.resetPages();
         }
```
Comment (ludoboludo) [inline assets/media-gallery.js:52]: I think we should do one other edit. I don't want the theme to scroll me back up to the active media if that active media is the same one as before. 
Below there is a `setTimeout()` in which we're doing the `window.scrollTo()`. We should add a check to not run that in the scenario that we're still showing the same media. 

Here is what I mean: [recording](https://screenshot.click/12-21-p8q9w-lxqsb.mp4)

Though to be fair it was also on previous versions... Maybe something we can double check with UX but I personally find it a bit annoying 🤔

**Review by ludoboludo — APPROVED** [review]
Looking good 👍 

Noted a potential extra tweak but it's a behaviour that's existing in previous version so not a high priority but a nice fix 🤷

File: assets/media-gallery.js (hunk 1 above)
Comment (lhoffbeck) [reply assets/media-gallery.js:52]: Yeah that seems valuable, maybe we roll it out as a separate pr?

**Review by Roi-Arthur — COMMENTED** [review]
The comment I Ieft is mostly me not knowing the syntax of the proposed change.
Seems good to go otherwise

**Pull request merged**